### PR TITLE
Feat: Database Backups

### PR DESCRIPTION
**Pull request by ItzNotABug**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

**Comment by ItzNotABug** [comment]
cc @TorstenDittmann @Meldiron @christyjacob4 @fogelito

File: src/lib/stores/database.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    // fast path: return if user dismissed the banner
+    if (!shouldShowNotification(backupsBannerId)) return;
+
+    let total = 0;
+
+    try {
+        const policies = await sdk.forProject.backups.listPolicies([
+            Query.limit(1),
+            Query.equal('resourceId', database.$id)
+        ]);
```
[the remark this answers is not in the document]
Comment (ItzNotABug) [reply src/lib/stores/database.ts:25]: We do but that requires a `policyId` and how'd we get the policy count then?

File: src/routes/(console)/project-[project]/databases/+page.ts [hunk 2]
```diff
@@ -1,24 +1,91 @@
...
+// TODO: @itznotabug we should improve this!
+async function fetchDatabasesAndBackups(limit: number, offset: number) {
+    const databases = await sdk.forProject.databases.list([
+        Query.limit(limit),
+        Query.offset(offset),
+        Query.orderDesc('')
+    ]);
+
+    const policies = await fetchPolicies(databases);
+    const lastBackups = await fetchLastBackups(databases);
```
[the remark this answers is not in the document]
Comment (ItzNotABug) [reply src/routes/(console)/project-[project]/databases/+page.ts:38]: Done.

File: src/lib/components/drop.svelte [hunk 3]
```diff
@@ -42,7 +42,10 @@
                 {
                     name: 'offset',
                     options: {
-                        offset: [noArrow ? 0 : -arrowSize, noArrow ? 0 : arrowSize / 1.5]
+                        offset:
+                            placement !== 'bottom-end'
+                                ? [noArrow ? 0 : -arrowSize, noArrow ? 0 : arrowSize / 1.5]
+                                : [noArrow ? 0 : arrowSize * 1.75, noArrow ? 0 : arrowSize / 1.5]
```
[the remark this answers is not in the document]
Comment (ItzNotABug) [reply src/lib/components/drop.svelte:48]: checked at a few places, didn't see any misalignment.

[the remark this answers is not in the document]
Comment (ItzNotABug) [reply src/lib/components/drop.svelte:48]: We removed the arrows from drop-downs, will revert this change.

File: src/lib/helpers/notifications.ts [hunk 4]
```diff
@@ -18,7 +18,8 @@ const userPreferences = () => get(user).prefs;
 
 const notificationPrefs = (): Record<string, NotificationPrefItem> => {
     const prefs = userPreferences();
-    return prefs.notificationPrefs ? prefs.notificationPrefs : {};
+    // for some reason, the prefs become array as default or on all clear. let's reset.
+    return Array.isArray(prefs.notificationPrefs) ? {} : prefs.notificationPrefs || {};
```
Comment (TorstenDittmann) [inline src/lib/helpers/notifications.ts:22]: thats actually a php downside, since for php everything is an array 👍🏻

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**